src/Tribe/Log/Admin.php

```diff
@@ -0,0 +1,249 @@
+<?php
+class Tribe__Log__Admin {
+	public function __construct() {
+		add_action( 'wp_ajax_tribe_logging_controls', array( $this, 'listen' ) );
+		add_action( 'init', array( $this, 'serve_log_downloads' ) );
+	}
+
+	/**
+	 * Returns the HTML comprising the event log section for use in the
+	 * Events > Settings > Help screen.
+	 *
+	 * @return string
+	 */
+	public function display_log() {
+		$log_choices  = $this->get_available_logs();
+		$log_engines  = $this->get_log_engines();
+		$log_levels   = $this->get_logging_levels();
+		$log_entries  = $this->get_log_entries();
+		$download_url = $this->get_log_url();
+
+		$this->setup_script();
+
+		ob_start();
+		include trailingslashit( Tribe__Main::instance()->plugin_path ) . 'src/admin-views/event-log.php';
+		return ob_get_clean();
+	}
+
+	/**
+	 * Listens for changes to the event log settings updating and returning
+	 * an appropriate response.
+	 */
+	public function listen() {
+		$fields = wp_parse_args( $_POST, array(
+			'check'      => '',
+			'log-level'  => '',
+			'log-engine' => '',
+		) );
+
+		foreach ( $fields as &$single_field ) {
+			$single_field = filter_var( $single_field, FILTER_SANITIZE_STRING, FILTER_FLAG_ENCODE_HIGH );
```

Comment:
shouldn't we use? [`sanitize_text_field`](https://codex.wordpress.org/Function_Reference/sanitize_text_field)